server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -27,12 +35,16 @@
  */
 public final class TaskExecutionTimeTrackingEsThreadPoolExecutor extends EsThreadPoolExecutor {
 
+    private static final int[] LATENCY_PERCENTILES_TO_REPORT = { 50, 90 };
```

Comment:
I think we should add 99 too. To some extent, the max-value in the interval seems most important.

I wonder if we also want an actual avg, but we can possibly add that.

At one point I was considering having the current value too. That would be easy to get, just peek into the queue for the front element and diff now-creation. But I think the percentiles or at least the 99th percentile also adds value so we can start there.

Reply:
Added p99 in d6e44edbe67c86cb0813936cb6c0f39d6483458a